### PR TITLE
Add Enrollment and Certificate Signing requests / approval functionality

### PR DESCRIPTION
**Pull request by DakCrowder**
Adds:

- Adds enrollmentrequests / csr endpoint for usage with the existing `flightctl` module
- New `flightctl_approve` module that allows for approving or denying requests
- Start writing some simple unit tests which can be run via `make test-unit`
- Start writing some integration tests which can be run via `make integration-test`
  - Currently, these are dependent on a local flightctl instance being run and aren't really configurable or able to be run as a part of a CI process.  The goal here is to begin writing them to aid in development and so that the roles/tests themselves will be in place for a more robust CI pipeline in the future

### CODE REVIEW

File: plugins/modules/flightctl_approve.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+DOCUMENTATION = r"""
+module: flightctl_approve
+short_description: Approve or deny requests
+author:
+  - "Dakota Crowder (@dakcrowder)"
+description:
+  - Approve or deny enrollment or certificate signing requests
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_approve.py:14]:
```suggestion
  - Approve or deny enrollment or certificate signing requests.
```

File: plugins/modules/flightctl_approve.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+options:
+  kind:
+    description:
+      - Use to specify an object model.
+    type: str
+    required: True
+  name:
+    description:
+      - Use to specify a device name
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_approve.py:23]:
```suggestion
      - Use to specify a device name.
```

File: plugins/modules/flightctl_approve.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    type: str
+  approved:
+    description:
+      - Indicates if the request should be approved (if True) or denied (if False)
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_approve.py:27]:
```suggestion
      - Indicates if the request should be approved (if True) or denied (if False).
```

File: plugins/modules/flightctl_approve.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    type: bool
+    required: True
+  approvedBy:
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_approve.py:30]:
```suggestion
  approved_by:
```

File: plugins/modules/flightctl_approve.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    description:
+      - Name of the approver
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_approve.py:32]:
```suggestion
      - Name of the approver.
```

File: plugins/modules/flightctl_approve.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    type: str
+  labels:
+    description:
+      - Labels that will be applied to the device on approval
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_approve.py:36]:
```suggestion
      - Labels that will be applied to the device on approval.
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,8 @@
+TEST_ARGS ?= ""
```
Comment (alinabuzachis) [inline Makefile:1]: You can also have a look at the tox.ini file https://github.com/flightctl/flightctl-ansible/blob/main/tox.ini

**Comment by alinabuzachis** [comment]
I also suggest checking whether the module is running in check_mode and whether idempotency is assured.

**Review by alinabuzachis — APPROVED** [review]

File: plugins/module_utils/constants.py [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# coding: utf-8 -*-
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
```
Comment (alinabuzachis) [inline plugins/module_utils/constants.py:4]: Does it make sense to move https://github.com/flightctl/flightctl-ansible/pull/4/files#diff-5610ae59ef2d2d3dd2b4c2c6f6714b3ca717790cf804603f03fcac3a34d3d162L84 here?

Comment (DakCrowder) [reply plugins/module_utils/constants.py:4]: I think its ok for the api module to own the endpoint definitions as it seems like the proper "owner" for that info and right now I don't think anything else requires that data.  

However i do think that given I've added definitions for two of the kind/resource types in the constants file maybe all of them should be defined there and possibly something like an enum could be used in the future to better tie up those values (and those could then be referenced in the API_ENDPOINTS definition).

**Pull request merged**